lahja/asyncio/endpoint.py

```diff
@@ -653,11 +523,22 @@ def is_connected_to(self, endpoint_name: str) -> bool:
 
         if event_type in self._sync_handler:
```

Comment:
Man, coroutines and mypy are pretty frustrating to work with. You have to write the exception handling logic twice for no good reason, and the naive way of getting around this is frustrated by mypy:

```python
async def call_handler(handler, item, is_sync: bool) -> None:
  try:
    if is_sync:
      handler(item)
    else:
      await handler(iteem)
  except Exception:
    self.logger.debug('Error in subscription handler")

if event_type in self._sync_handler:
    for handler in self._sync_handler[event_type]:
      await call_handler(handler, item, is_sync=True)

if event_type in self._async_handler:
  await asyncio.gather(
    call_handler(handler, item, is_sync=False)
    for handler in self._async_handler[event_type]
  )
```